components/ai_chat/core/browser/utils_unittest.cc

```diff
@@ -5,24 +5,202 @@
 
 #include "brave/components/ai_chat/core/browser/utils.h"
 
+#include <utility>
+
+#include "base/strings/string_number_conversions.h"
+#include "brave/components/ai_chat/core/browser/constants.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom.h"
+#include "testing/gmock/include/gmock/gmock.h"
 #include "testing/gtest/include/gtest/gtest.h"
 #include "url/gurl.h"
 
 namespace ai_chat {
 
-TEST(AIChatUtilsUnitTest, IsBraveSearchSERP) {
+// Test fixture class for AI Chat utility functions
+class AIChatUtilsUnitTest : public ::testing::Test {
+ protected:
+  // Setup or helper methods could go here if needed in the future
+};
+
+// Test IsBraveSearchSERP function with various GURL inputs
+TEST_F(AIChatUtilsUnitTest, IsBraveSearchSERP) {
   EXPECT_TRUE(IsBraveSearchSERP(GURL("https://search.brave.com/search?q=foo")));
-  // Missing or wrong path.
+
+  // Missing or wrong path
   EXPECT_FALSE(IsBraveSearchSERP(GURL("https://search.brave.com?q=foo")));
   EXPECT_FALSE(
       IsBraveSearchSERP(GURL("https://search.brave.com/test.html?q=foo")));
-  // Missing or wrong query parameter.
+
+  // Missing or wrong query parameter
   EXPECT_FALSE(IsBraveSearchSERP(GURL("https://search.brave.com/search")));
   EXPECT_FALSE(IsBraveSearchSERP(GURL("https://search.brave.com/search?t=t")));
-  // HTTP scheme.
+
+  // HTTP scheme instead of HTTPS
   EXPECT_FALSE(IsBraveSearchSERP(GURL("http://search.brave.com/search?q=foo")));
-  // Wrong host.
+
+  // Wrong host
   EXPECT_FALSE(IsBraveSearchSERP(GURL("https://brave.com/search?q=foo")));
 }
 
+// GetMaxPageContentLengthFromContextSize with various context sizes
+TEST_F(AIChatUtilsUnitTest, GetMaxPageContentLengthFromContextSize) {
+  struct TestCase {
+    std::optional<uint32_t> context_size;
+    uint32_t expected_max_content_length;
+  };
+
+  TestCase test_cases[] = {
+      {1000, 1000 * kDefaultCharsPerToken},
+      {4000, 4000 * kDefaultCharsPerToken},
+      {8000, 8000 * kDefaultCharsPerToken},
+      // No context size provided, so Brave should use the default context size
+      {std::nullopt, kCustomModelDefaultContextSize * kDefaultCharsPerToken},
+      // Edge cases
+      {0, 0},
+      {1, 1 * kDefaultCharsPerToken},
+  };
```

Comment:
please tag this as `static constexpr` so we know for sure there is no undefined behaviour in these multiplications, as the compiler will check them.